### PR TITLE
6352 packages bsd

### PR DESCRIPTION
**Pull request by konangc87**
|Related issue|
|---|
|#6352|

## **Description**

This issue aims to make use of the generic synchronization libraries, to use the synchronization solution algorithm that both shared modules provide.

These changes have to be made in the various implementations according to the operating system targeted for this feature.

## **DoD**
- [X] Implementation
- [X] **Run testtool on MAC**
image
- [X] **Run testtool on FreeBSD**
image
- [X] **RTR
image
- **SysInfo UTs:**
> **Linux:**
image
> **Windows:**
image

### CODE REVIEW

File: src/shared_modules/utils/tests/filesystemHelper_test.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+void FilesystemUtilsTest::TearDown() {};
+#ifdef WIN32
+TEST_F(FilesystemUtilsTest, FilesystemExistsDir)
+{
+    EXPECT_TRUE(Utils::existsDir(R"(C:\)"));
+}
+
+TEST_F(FilesystemUtilsTest, FilesystemEnumerateDir)
+{
+	const auto items {Utils::enumerateDir(R"(C:\)")};
```
Comment (tinachoB) [inline src/shared_modules/utils/tests/filesystemHelper_test.cpp:26]: wrong indentation

File: src/shared_modules/utils/tests/filesystemHelper_test.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+#else
+TEST_F(FilesystemUtilsTest, FilesystemExistsDir)
+{
+    EXPECT_TRUE(Utils::existsDir(R"(/usr)"));
+}
+
+TEST_F(FilesystemUtilsTest, FilesystemEnumerateDir)
+{
+	const auto items {Utils::enumerateDir(R"(/usr)")};
```
Comment (tinachoB) [inline src/shared_modules/utils/tests/filesystemHelper_test.cpp:38]: ditto

**Review by tinachoB — CHANGES_REQUESTED** [review]
Nice work

**Review by Dwordcito — APPROVED** [review]
looks good

**Review by tinachoB — APPROVED** [review]

**Review by tinachoB — APPROVED** [review]

**Pull request merged**